//
// Copyright 2013-2015, Andr√© Puschmann <andre.puschmann@tu-ilmenau.de>
//
// This file is part of libgdtp.
//
// This is free software; you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation; either version 3, or (at your option)
// any later version.
//
// This software is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this software; see the file COPYING.  If not, write to
// the Free Software Foundation, Inc., 51 Franklin Street,
// Boston, MA 02110-1301, USA.

message GdtpPdu {
  enum PduType {
    DATA = 0;
    ACK = 1;
    BROADCAST = 2;
  }
  required uint32 src_id = 1;
  required uint32 dest_id = 2;
  required uint64 source = 3;
  required uint64 destination = 4;
  optional uint64 seqno = 5;
  required PduType type = 6;
  repeated bytes payload = 7;
}

message GdtpFrame {
  repeated GdtpPdu pdu = 1;
}
